test/test_piece_picker.cpp

```diff
@@ -885,64 +885,72 @@ TORRENT_TEST(cursors_sweep_in_we_have)
-	{
-		p->we_dont_have(i);
-		TEST_EQUAL(p->cursor(), piece_index_t(0));
-		TEST_EQUAL(p->reverse_cursor(), next(i));
-	}
+	auto p = setup_picker("7654321", "       ", "", "");
 	TEST_CHECK(!p->is_finished());
 	TEST_CHECK(!p->is_seeding());
 	TEST_EQUAL(p->cursor(), piece_index_t(0));
 	TEST_EQUAL(p->reverse_cursor(), piece_index_t(7));
+	p->set_sequential_range(piece_index_t(1), piece_index_t(5));
+	for (piece_index_t i(1); i < piece_index_t(6); ++i)
+	{
+		TEST_EQUAL(p->cursor(), piece_index_t(i));
+		TEST_EQUAL(p->reverse_cursor(), piece_index_t(6));
+		p->we_have(i);
+	}
+	TEST_EQUAL(p->cursor(), piece_index_t(7));
+	TEST_EQUAL(p->reverse_cursor(), piece_index_t(0));
+	p->we_have(piece_index_t(0));
+	p->we_have(piece_index_t(6));
+	TEST_CHECK(p->is_finished());
+	TEST_CHECK(p->is_seeding());
 }
 
-TORRENT_TEST(cursors_sweep_down_we_dont_have)
+TORRENT_TEST(cursors_sweep_down_seq_range_we_have)
 {
-	auto p = setup_picker("7654321", "*******", "", "");
-	TEST_CHECK(p->is_finished());
-	TEST_CHECK(p->is_seeding());
-	TEST_EQUAL(p->cursor(), piece_index_t(7));
-	TEST_EQUAL(p->reverse_cursor(), piece_index_t(0));
-	for (piece_index_t i(6); i >= piece_index_t(0); --i)
-	{
-		p->we_dont_have(i);
-		TEST_EQUAL(p->cursor(), i);
-		TEST_EQUAL(p->reverse_cursor(), piece_index_t(7));
-	}
+	auto p = setup_picker("7654321", "       ", "", "");
 	TEST_CHECK(!p->is_finished());
 	TEST_CHECK(!p->is_seeding());
 	TEST_EQUAL(p->cursor(), piece_index_t(0));
 	TEST_EQUAL(p->reverse_cursor(), piece_index_t(7));
+	p->set_sequential_range(piece_index_t(1), piece_index_t(5));
+	for (piece_index_t i(5); i >= piece_index_t(1); --i)
+	{
+		TEST_EQUAL(p->cursor(), piece_index_t(1));
+		TEST_EQUAL(p->reverse_cursor(), next(i));
+		p->we_have(i);
+	}
+	TEST_EQUAL(p->cursor(), piece_index_t(7));
+	TEST_EQUAL(p->reverse_cursor(), piece_index_t(0));
+	p->we_have(piece_index_t(0));
+	p->we_have(piece_index_t(6));
+	TEST_CHECK(p->is_finished());
+	TEST_CHECK(p->is_seeding());
 }
 
-TORRENT_TEST(cursors_sweep_out_we_dont_have)
+TORRENT_TEST(cursors_sweep_in_seq_range_we_have)
 {
-	auto p = setup_picker("7654321", "*******", "", "");
-	TEST_CHECK(p->is_finished());
-	TEST_CHECK(p->is_seeding());
-	TEST_EQUAL(p->cursor(), piece_index_t(7));
-	TEST_EQUAL(p->reverse_cursor(), piece_index_t(0));
-	for (piece_index_t left(3), right(3);
-		left >= piece_index_t(0) && right < piece_index_t(7);
-		--left, ++right)
-	{
-		p->we_dont_have(left);
-		p->we_dont_have(right);
-		TEST_EQUAL(p->cursor(), left);
-		TEST_EQUAL(p->reverse_cursor(), next(right));
-	}
+	auto p = setup_picker("7654321", "       ", "", "");
 	TEST_CHECK(!p->is_finished());
 	TEST_CHECK(!p->is_seeding());
 	TEST_EQUAL(p->cursor(), piece_index_t(0));
 	TEST_EQUAL(p->reverse_cursor(), piece_index_t(7));
+	p->set_sequential_range(piece_index_t(1), piece_index_t(5));
+	for (piece_index_t left(1), right(5); left <= piece_index_t(3)
+		&& right >= piece_index_t(3); ++left, --right)
```

Comment:
instead of comparing `right` and `left` against 3, it seems simpler to just say `left < right`

Reply:
`left <= right`